OpenRA.Game/Server/Server.cs

```diff
@@ -308,9 +308,9 @@ void ValidateClient(Connection newConn, string data)
 					Slot = LobbyInfo.FirstEmptySlot(),
 					PreferredColor = handshake.Client.Color,
 					Color = handshake.Client.Color,
-					Faction = "Random",
+					Faction = handshake.Client.Faction,
```

Comment:
The faction is already validated after loading it from settings. Isn't it sufficient? I think also now it isn't validated each time you change Faction.